src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -14,59 +14,102 @@
 limitations under the License.
 -->
 
-<md-input-container class="md-block">
-  <label>Container image</label>
-  <input ng-model="ctrl.containerImage" required>
-</md-input-container>
-<md-input-container class="md-block">
-  <label>Number of pods</label>
-  <input ng-model="ctrl.replicas" type="number" required min="1">
-</md-input-container>
-<div layout="row" ng-repeat="portMapping in ctrl.portMappings">
-  <md-input-container flex>
-    <label>Port</label>
-    <input ng-model="portMapping.port" type="number" min="0">
+<kd-help-section>
+  <md-input-container class="md-block">
+    <label>Container image</label>
+    <input ng-model="ctrl.containerImage" required>
   </md-input-container>
-  <md-input-container flex>
-    <label>Target port</label>
-    <input ng-model="portMapping.targetPort" type="number" min="0">
+  <kd-user-help>
+    Enter the URL of a public image on any registry, or a private image hosted on Docker Hub or Google Container Registry
+    <a href=""> Learn More </a>
+  </kd-user-help>
+</kd-help-section>
+
+<kd-help-section>
+  <md-input-container class="md-block">
+    <label>Number of pods</label>
+    <input ng-model="ctrl.replicas" type="number" required min="1">
   </md-input-container>
-  <md-input-container flex="none">
-    <label>Protocol</label>
-    <md-select ng-model="portMapping.protocol" required>
-      <md-option ng-repeat="protocol in ctrl.protocols" ng-value="protocol">
-        {{protocol}}
-      </md-option>
-    </md-select>
+  <kd-user-help>
+    A Replica Set will be created to maintain the desired number of pods across your cluster
+    <a href=""> Learn More </a>
+  </kd-user-help>
+</kd-help-section>
+
+<kd-help-section>
+  <div layout="row" ng-repeat="portMapping in ctrl.portMappings">
+    <md-input-container flex>
+      <label>Port</label>
+      <input ng-model="portMapping.port" type="number" min="0">
+    </md-input-container>
+    <md-input-container flex>
+      <label>Target port</label>
+      <input ng-model="portMapping.targetPort" type="number" min="0">
+    </md-input-container>
+    <md-input-container flex="none">
+      <label>Protocol</label>
+      <md-select ng-model="portMapping.protocol" required>
+        <md-option ng-repeat="protocol in ctrl.protocols" ng-value="protocol">
+          {{protocol}}
+        </md-option>
+      </md-select>
+    </md-input-container>
+  </div>
+  <kd-user-help>
+    Ports are optional. If specified, a Service will be created mapping the Port (incoming)to a target Port seen by the container. The internal DNS name for this Service will be webserver
```

Comment:
`Will be webserver`? I dont think so :) This needs a binding to internal endpoints or human readable description, like, `The internal DNS name for this Service will be <name>.<namespace>`. 

PS. Finish sentence with dot.